test/distributions/test_constraints.py

```diff
@@ -113,5 +113,5 @@ def test_transform_to(constraint_fn, args, is_cuda):
     assert torch.allclose(y, y2), "Error in transform_to({}) pseudoinverse".format(constraint)
 
 
-if __name__ == '__main__':
+if __name__ == '__main__' and torch._C.has_lapack:
```

Comment:
If someone runs `python test/distributions/test_constraints.py` without LAPACK support in their compilation, this would just silently not run the tests without explanation, no?

Reply:
@janeyx99 Yep. [Proposed](https://github.com/pytorch/pytorch/pull/70671#discussion_r779730678) in the discussion above by @neerajprad . It mimics the approach found in another module. It's a secondary protection, the main change is the fix to `run_tests.py`. But after your mention of the silentness of the test I started thinking is this a good idear or we should either get rid of it, or turn this into a more verbose and graceful failure. @neerajprad  WDYT

Reply:
I am fine with removing this check as well (if it doesn't break anything else). My initial impression was that the absence of this check was the cause for failures in `test_constraints`, but since that's not the case, there is no reason to keep it. The resulting test failures if a user runs these tests directly should be self explanatory.

Reply:
@janeyx99 @neerajprad pushed

Reply:
LGTM, thanks @lithuak.

Reply:
@janeyx99 - Thanks for reviewing, bumping this up for a final review.